templates/repo/clone_script.tmpl

```diff
@@ -17,7 +17,13 @@
 		const btn = isSSH ? sshBtn : httpsBtn;
 		if (!btn) return;
 
-		const link = btn.getAttribute('data-link');
+		let link = btn.getAttribute('data-link');
+		if (link.startsWith('http://') || link.startsWith('https://')) {
+			// use current host as the clone link
+			const url = new URL(link);
+			url.host = window.location.host;
+			link = url.toString();
+		}
```

Comment:
```suggestion
		let link = btn.getAttribute('data-link');
		if (link.startsWith('http')) {
			// use current host as the clone link
			const url = new URL(link);
			url.origin = window.location.origin;
			link = url.toString();
		}
```

Protocol can be different as well so `origin` replacement is better.

Reply:
Could also use `parseUrl` from `utils` to convert URL to absolute and skip the `if` clause, but as we can not `import` in this script, the function would need to be duplicated.

Reply:
`link.startsWith('http')` is not always right. Assume the url is a ssh one with `http_my_gitea.a.com`?

Reply:
Updated to `/https?:/.test(link)`

Reply:
I do not like the regex. IMO `startsWith('http://')` is clean enough and faster.

Reply:
And, it's wrong to change `origin`

https://developer.mozilla.org/en-US/docs/Web/API/URL/origin

> The origin read-only property of the [URL](https://developer.mozilla.org/en-US/docs/Web/API/URL) interface returns a string containing the Unicode serialization of the origin of the represented URL.

Reply:
> And, it's wrong to change origin

Interesting, it definitely is not read-only as that document says, but probably better to do it your way.

Reply:
That's standard. Following standards would reduce unnecessary bugs.

And there is a strong reason about why it is readonly: the `origin` only contains protocol and host, it doesn't contain username and password. If the `origin` could be changed, the the username & password would be messy.

```js
u = new URL("http://a:b@c/d") 
// now u.origin is "http://c/", it's unclear if it could be assigned to something else.
```